File: test_regress/t/t_lib_nolib.pl
#!/usr/bin/env perl
# Makes the test run with tracing enabled by default, can be overridden
# with --notrace
unshift(@ARGV, "--trace");
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2019 by Todd Strader. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

scenarios(
    vlt => 1,
    xsim => 1,
    );

$Self->{sim_time} = $Self->{benchmark} * 100 if $Self->{benchmark};
top_filename("t/t_lib_prot.v");

# Tests the same code as t_lib_prot.pl but without --protect-lib
compile(
    verilator_flags2 => ["t/t_lib_prot_secret.v"],
    xsim_flags2 => ["t/t_lib_prot_secret.v"],
    );

execute(
    check_finished => 1,
    );

if ($Self->{vlt} && $Self->{trace}) {
    # We can see the ports of the secret module
    file_grep("$Self->{obj_dir}/simx.vcd", qr/accum_in/);
    # and we can see what's inside (because we didn't use --protect-lib)
    file_grep("$Self->{obj_dir}/simx.vcd", qr/secret_/);
}

ok(1);
1;

File: test_regress/t/t_lib_nolib.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('vlt_all', 'xsim')
test.top_filename = "t/t_lib_prot.v"

if test.benchmark:
    test.sim_time = test.benchmark * 100

trace_opt = ("" if re.search(r'--no-trace', ' '.join(test.driver_verilator_flags)) else "-trace")

# Tests the same code as t_lib_prot.py but without --protect-lib
test.compile(verilator_flags2=['--no-timing', trace_opt, "t/t_lib_prot_secret.v"],
             xsim_flags2=["t/t_lib_prot_secret.v"])

test.execute()

if test.vlt and test.trace:
    # We can see the ports of the secret module
    test.file_grep(test.trace_filename, r'accum_in')
    # and we can see what's inside (because we didn't use --protect-lib)
    test.file_grep(test.trace_filename, r'secret_')

test.passes()
